tools/infrasys/src/shared.rs

```diff
@@ -0,0 +1,91 @@
+use log::info;
+use rusoto_cloudformation::{CloudFormation, CloudFormationClient, DescribeStacksInput, Parameter};
+use snafu::{ensure, OptionExt, ResultExt};
+use std::{env, thread, time};
+use structopt::StructOpt;
+
+use super::{error, Result};
+
+#[derive(Debug, StructOpt)]
+pub enum KeyRole {
+    Root,
+    Publication,
+}
+
+/// Retrieve a BUILDSYS_* variable that we expect to be set in the environment
+pub fn getenv(var: &str) -> Result<String> {
+    env::var(var).context(error::Environment { var })
+}
+
+/// Generates a parameter type object used to specify parameters in CloudFormation templates
+pub fn get_parameter(key: String, val: String) -> Parameter {
```

Comment:
Nit: maybe call it `create_parameter` or `make_parameter`, or `new_parameter`.
 
```suggestion
pub fn create_parameter(key: String, val: String) -> Parameter {
```